pytest_twisted.py

```diff
@@ -1,17 +1,30 @@
 import functools
 import inspect
 import sys
+import traceback
 import warnings
 
 import decorator
 import greenlet
 import pytest
+import six
 
 from twisted.internet import error, defer
 from twisted.internet.threads import blockingCallFromThread
 from twisted.python import failure
 
 
+class AddReaderNotImplementedError(Exception):
+    @classmethod
+    def build(cls):
+        return cls(
+            "Failed to install asyncio reactor.  The proactor was"
+            " used and is lacking the needed `.add_reader()` method"
+            " as of Python 3.8 on Windows."
+            "  https://twistedmatrix.com/trac/ticket/9766"
```

Comment:
This error message could use some love. The proactor doesn't implement `add_reader`, period, not just on 3.8.

More to the point, why the single-case exception? I think it would be better to just raise `NotImplementedError` with that message in the `except` block.

Reply:
The proactor might implement `.add_reader()` in the future.  But sure, we know for <= 3.8.1.

I like specific exceptions and not incidentally catching those I don't intend by libraries providing their own. Look at the hoops to figure out if it is the `NotImplementedError` from `.add_reader()`.  But I did consider inheriting from `NotImplementedError` since it is a `raise from` situation.